lib/Doctrine/DBAL/Cache/ArrayStatement.php

```diff
@@ -130,7 +131,14 @@ public function fetchColumn($columnIndex = 0)
     {
         $row = $this->fetch(FetchMode::NUMERIC);
 
-        // TODO: verify that return false is the correct behavior
-        return $row[$columnIndex] ?? false;
+        if ($row === false) {
+            return false;
+        }
+
+        if ($columnIndex >= count($row)) {
```

Comment:
Would an `array_key_exists()` be better here?

Reply:
Actually, yes. Otherwise, the case when `$columnIndex < 0` won't be handled.